content/departments/cloud/technical-docs/private-code-hosts/index.md

```diff
@@ -0,0 +1,153 @@
+# Managed Instance Private Code Hosts support
+
+> **Warning**
+> This is still work in progress!
+
+Every [v2.0 Cloud Instance](../v2.0/index.md) is deployed in Google Cloud Platform.
+
+**Private Code Host** is a code host deployed in a private network (for example AWS EC2 instance within VPC). To connect to this code host a user has to have access to the private network usually via VPC Peering, VPN, or tunneling.
+
+## Managed Instance NAT IP allowlist solution (recommended)
+
+This solution is recommended by Cloud Team, but only possible if customers' code hosts can be accessible publicly and customer is able to allow incoming traffic from Sourcegraph-owned static IP addresses.
+Outgoing traffic of Cloud instances goes through Cloud NAT with stable IPs. All IPs are reservered exclusively on a per customer basis.
+[More informations about IP allowlist](../../index.md#faq-what-is-the-cloud-instance-ip)
+
+If customer is not able to meet these requirements, Sourcegraph can propose [VPN solution for code hosts deployed on AWS](#aws-gcp-site-to-site-vpn-solution)
+
+## AWS GCP site to site VPN solution (AWS code hosts only)
+
+### Architecture
```

Comment:
```suggestion
## AWS GCP site-to-site VPN solution (AWS code hosts only)

This option is for customers who want to connect to a private code host that is hosted on AWS infrastructure.

### Architecture
```

In case we want to add a section in the future for private code hosts hosted in GCP